+sw_tests/+unit_tests/unittest_spinw.m

```diff
@@ -0,0 +1,119 @@
+classdef unittest_spinw < matlab.mock.TestCase
+    % Runs through unit tests for @spinw/spinw.m
+    properties (TestParameter)
+        spinw_struct_input = { ...
+            {struct('lattice', struct('angle', [pi, pi, (2*pi)/3], ...
+                                     'lat_const', [2, 2, 4])),
+             'spinw_from_struct_lat224.mat'}, ...
+            {struct('lattice', struct('angle', [pi; pi; (2*pi)/3], ...
+                                     'lat_const', [2; 2; 4])), ...
+             'spinw_from_struct_lat224.mat'}};
+        spinw_figure_input = { ...
+            {'default_structure.fig', 'spinw_default.mat'}, ...
+            {'afm_chain_spec.fig', 'spinw_afm_chain.mat'}}
+        spinw_file_input = { ...
+            {'YFeO3_mcphase.cif', 'spinw_from_cif_YFeO3_mcphase.mat'}, ...
+            {'LaFeO3_fullprof.cif', 'spinw_from_cif_LaFeO3_fullprof.mat'}, ...
+            {'BiMn2O5.fst', 'spinw_from_fst_BiMn2O5.mat'}};
+    end
+    methods (Static)
+        function udir = get_unit_test_dir()
+            udir = fullfile('.', 'test_data', 'unit_tests');
+        end
+    end
+    methods
+        function obj = load_spinw(testCase, filename)
+            obj = load(fullfile(testCase.get_unit_test_dir(), 'spinw', filename));
+            obj = obj.data;
+        end
+        function obj = load_figure(testCase, filename)
+            path = fullfile(testCase.get_unit_test_dir(), 'Figure', filename);
+            obj = openfig(path, 'invisible');
+        end
+        function verify_obj(testCase, expected_obj, actual_obj, rel_tol, abs_tol)
+            if nargin < 5
+                abs_tol = 0;
+            end
+            if nargin < 4
+                rel_tol = 1e-10;
+            end
+            all_fieldnames = union(fieldnames(expected_obj), fieldnames(actual_obj));
+            for i=1:length(all_fieldnames)
+                field = all_fieldnames(i);
+                if strcmp(field{:}, "cache")
+                    continue;
+                end
+                expected_value = expected_obj.(field{:});
+                actual_value = actual_obj.(field{:});
+                if isstruct(expected_value)
+                    testCase.verify_obj(expected_value, actual_value, rel_tol, abs_tol);
+                else
+                    testCase.verify_val(expected_value, actual_value, rel_tol, abs_tol, field{:});
+                end
+            end
+        end
+        function verify_val(testCase, expected_val, actual_val, rel_tol, abs_tol, field)
```

Comment:
Feels like `verify_val` (and perhaps `verify_obj`) should be a utility function much like `mock_function`  - I can imagine wanting to use these in future unit tests.

Reply:
Yes! I just put them here for now since we don't have any other test files yet, but I can move them out as part of this PR if you want to use them.

Reply:
If you don't mind I think it makes sense to do it in this PR to avoid merge conflicts in subsequent PRs as we'll both be working on unit tests and probably both need it as a utility function!